user-journey-tests/TB/helpers/testHelpers/receivingLicence.js

```diff
@@ -10,24 +13,60 @@ import { navigateToTaskList } from './taskListNav.js'
 const defaultEmail = 'eoin.corr@esynergy.co.uk'
 const defaultFirstName = 'Bruce'
 const defaultLastName = 'Wayne'
+const defaultYourFirstName = 'Clark'
+const defaultYourLastName = 'Kent'
+const defaultOriginEmail = 'origin@esynergy.co.uk'
+const defaultDestinationEmail = 'eoin.corr@esynergy.co.uk'
 
 // Helper function to complete the origin task
 const completeLicenceTask = async ({
   email = defaultEmail,
   firstName = defaultFirstName,
   lastName = defaultLastName,
+  yourFirstName = defaultYourFirstName,
+  yourLastName = defaultYourLastName,
+  originEmail = defaultOriginEmail,
+  destinationEmail = defaultDestinationEmail,
   on = false
 } = {}) => {
   await navigateToTaskList()
   await taskListPage.selectReceiveTheLicence(ownerNamePage)
-  await ownerNamePage.inputTextAndContinue(firstName, lastName, emailPage)
-  await emailPage.inputTextAndContinue(email, licenceAnswersPage)
+
+  // When on-farm with TB restricted origin and destination, the flow includes yourNamePage, originEmailAddressPage, and destinationEmailAddressPage
+  if (on) {
```

Comment:
Check with @eoin-corr-git  if he can have a look at this and all the journey test updates in this PR please